spec/helpers/blacklight_helper_spec.rb

```diff
@@ -128,4 +150,24 @@
       expect(helper).to have_received(:head).with(:bad_request)
     end
   end
+
+  describe "#link_back_to_catalog_safe" do
+    context "with valid parameters" do
+      let(:search_state) { FakeSearchState.new }
+      let(:current_search_session) { FakeValidCurrentSearchSession.new }
```

Comment:
Looking at this closer with Hector because of his question about these mocks. The mocks themselves seem fine, although there is likely some more elegant way of doing them, but these let statements are really confusing me a lot. They seem to be overriding methods in the actual helper namespace -- they're not used anywhere in the tests themselves. I don't really understand how this is the case and it makes me pretty uncomfortable.

I think we could try something like `allow(Blacklight::UrlHelper).to receive(:find_search_session).and_return(valid_session)`